src/pages-and-resources/data/thunks.js

```diff
@@ -19,6 +29,9 @@ export function fetchCourseApps(courseId) {
     try {
       const courseApps = await getCourseApps(courseId);
 
+      // eslint-disable-next-line max-len
+      courseApps.sort((firstEl, secondEl) => COURSE_APPS_ORDER.indexOf(firstEl.id) - COURSE_APPS_ORDER.indexOf(secondEl.id));
+
```

Comment:
You can use the following to avoid the line length issue. 
```suggestion
      courseApps.sort((firstEl, secondEl) => (
          COURSE_APPS_ORDER.indexOf(firstEl.id) - COURSE_APPS_ORDER.indexOf(secondEl.id)
      ));
```

I don't think this code will work as expected though. `addModels` in the next line will convert this to a map of the course app ID to the course app data, so the original ordering here will no longer matter. 

The sorting either needs to happen during read, or, the generic model store code can be updated to make it order-aware, i.e. by storing the order in which the ids should be traversed. 

Additionally, the above code will cause issues in case you have a custom course app plugin installed since that won't be listed here. Such apps will appear first, since the index of missing elements is -1. I'd recommend using an object that maps the app ids to their "priority" or order in the list.

Reply:
I have tested this solution. the Model is storing data as passed in sorted order so the first point is invalid here we are getting apps in sorted order without any issue. however, the second point is valid and things may get disturbed if some plugin id is missig it will be listed first

Reply:
@mehaknasir Objects keys in javascript may preserve insertion order, but this is not part of the spec, so it is definitely possible that the results will vary depending on browsers, or browser version.

Reply:
@xitij2000 all [browsers](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys) supports Objects keys regardless of browser version. The same goes for sort. only the internet explorer (5.0) version does not fully support the sort but the current explorer working version is 11.0. So I do not think so I would be an issue right now. I tested these changes in multiple browsers and this solution this working fine.
I agree with @mehaknasir on point 2. Things will get disturbed if some plugin id is missing it. Using the current solution, we need to add a plugin id in the frontend whenever a developer adds a new plugin in the backend. 
Your suggestions are welcomed. Let us know if you have any good solutions in your mind.

Reply:
@awais-ansari  Fair enough. The issue I was mentioning wasn't that sort and Object.keys aren't supported, but rather, if you insert items into a dictionary or object do you get them back in the same order. If you sort items and then put them in an object, they might lose their object *unless* objects guarantee that the insertion order is maintained. Perhaps my concern here is unfounded if you've found that it works regardless. 

I did mention a solution here, which is to use an object that maps each app id to its sort priority, and then any app not found can default to a value that will sort at the end.